File: frontend/src/api/notification.ts
/**
 * 通知 API
 */
import request from './index'

export interface NotificationItem {
    id: string
    type: string
    title: string
    content: string | null
    is_read: boolean
    related_id: string | null
    created_at: string
}

export interface NotificationList {
    notifications: NotificationItem[]
    total: number
    unread_count: number
}

/** 获取通知列表 */
export function getNotifications(params?: {
    skip?: number
    limit?: number
    unread_only?: boolean
}): Promise<NotificationList> {
    return request({ url: '/api/v1/user/notifications', method: 'get', params })
}

/** 获取未读数 */
export function getUnreadCount(): Promise<{ unread_count: number }> {
    return request({ url: '/api/v1/user/notifications/unread-count', method: 'get' })
}

/** 标记单条已读 */
export function markAsRead(id: string): Promise<{ detail: string }> {
    return request({ url: `/api/v1/user/notifications/${id}/read`, method: 'put' })
}

/** 全部已读 */
export function markAllRead(): Promise<{ detail: string }> {
    return request({ url: '/api/v1/user/notifications/read-all', method: 'put' })
}

File: backend/app/schemas/notification.py
"""
通知 Pydantic Schemas
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class NotificationItem(BaseModel):
    """单条通知"""
    id: str
    type: str
    title: str
    content: Optional[str] = None
    is_read: bool
    related_id: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class NotificationList(BaseModel):
    """通知列表响应"""
    notifications: list[NotificationItem]
    total: int
    unread_count: int
